### PR TITLE
[CGPROD-2795] responsive scrollable list

### PR DESCRIPTION
**Pull request by BeeGodwin**
- Scrollable list now consistently scales to fit the safe area on resize (if it's not right, then we need to update safe area)
- Inner padding is now consistent across any & all sizes
- Resolved a race condition that sometimes broke layout on resize
- Buttons now check that their click was within panel bounds before handling (gel buttons didn't respect rexUI's masking.)

### CODE REVIEW

File: test/core/layout/scrollable-list/scrollable-list.test.js [hunk 1]
```diff
@@ -40,75 +47,94 @@ const mockScene = {
     layout: {
         getSafeArea: jest.fn().mockReturnValue({ width: 100, height: 100 }),
     },
+    scale: { on: jest.fn() },
 };
 const mockGelButton = { width: 100, setScale: jest.fn() };
 buttons.createGelButton = jest.fn().mockReturnValue(mockGelButton);
+buttons.scaleButton = jest.fn();
 
 describe("Scrollable List", () => {
```
Comment (BeeGodwin) [inline test/core/layout/scrollable-list/scrollable-list.test.js:56]: Urgh, changing the nesting has brought most of this test suite into the diff.

File: src/core/layout/scrollable-list/scrollable-list-buttons.js [hunk 2]
```diff
@@ -39,9 +39,9 @@ const createGelButton = (scene, item) => {
...
     const { scene, config, gelButton } = args;
     const safeArea = scene.layout.getSafeArea();
-    const scaleFactor = (safeArea.width - config.space * 2) / gelButton.width;
+    const scaleFactor = (safeArea.width - config.space * 4) / gelButton.width;
     gelButton.setScale(scaleFactor);
     return args;
 };
 
-export { createGelButton };
+export { createGelButton, scaleButton };
```
Comment (AdamXA) [inline src/core/layout/scrollable-list/scrollable-list-buttons.js:47]: @FostUK are we doing exports like this or `export const`?

Comment (AnthonyGoodchild) [reply src/core/layout/scrollable-list/scrollable-list-buttons.js:47]: it's the same thing, no? just different syntax

Comment (FostUK) [reply src/core/layout/scrollable-list/scrollable-list-buttons.js:47]: That's only one export that's an object with createGelButton and scaleButton methods.

Comment (FostUK) [reply src/core/layout/scrollable-list/scrollable-list-buttons.js:47]: Kinda weird that it was an object with one method before :)

File: src/core/layout/scrollable-list/scrollable-list-helpers.js [hunk 3]
```diff
@@ -4,12 +4,21 @@
  * @author BBC Children's D+E
  * @license Apache-2.0 Apache-2.0
  */
-
 /* eslint-disable no-console */
-export const onClick = gelButton => {
-    console.log(`Clicked ${gelButton.config.id}`); // placeholder
-};
 
-export const assetKey = (key, assetKeys) => {
```
Comment (FostUK) [inline src/core/layout/scrollable-list/scrollable-list-helpers.js:13]: aNY REASON YOU ARE EXPORTING THIS TWICE?

File: src/core/layout/scrollable-list/scrollable-list-helpers.js [hunk 4]
```diff
@@ -4,12 +4,21 @@
...
-    return [assetKeys.prefix, key].join(".");
+/* gelButton's clickable zone extends beyond rexUI's mask [CGPROD-2795]*/
```
Comment (FostUK) [inline src/core/layout/scrollable-list/scrollable-list-helpers.js:9]: Prob don't need this comment - if there's anything that's a bit of a workaround we usually make sure there's a described test for it.

**Review by FostUK — APPROVED** [review]
Looks great! Added a minor comment.
Also **For future ref:** we try to avoid "helpers" files as they are a tiny bit of a code smell due to not really being descriptive and attract "code lint. You may have seen this somewhere else but it was a previous grad's work :)

Don't worry about it here though - merge merge merge!

**Pull request merged**